types/middlewares.d.ts

```diff
@@ -5,6 +5,7 @@ declare module 'mongoose' {
   type MongooseDistinctDocumentMiddleware = 'save' | 'init' | 'validate';
   type MongooseDocumentMiddleware = MongooseDistinctDocumentMiddleware | MongooseQueryAndDocumentMiddleware;
 
+  type MongooseRawResultQueryMiddleware = 'findOneAndUpdate' | 'findOneAndReplace' | 'findOneAndDelete';
   type MongooseDistinctQueryMiddleware = 'count' | 'estimatedDocumentCount' | 'countDocuments' | 'deleteMany' | 'distinct' | 'find' | 'findOne' | 'findOneAndDelete' | 'findOneAndReplace' | 'findOneAndUpdate' | 'replaceOne' | 'updateMany';
```

Comment:
shouldnt `MongooseDistinctQueryMiddleware` inherit `MongooseRawResultQueryMiddleware` instead of defining it duplicately?

Reply:
While `MongooseDistinctQueryMiddleware` is a superset of `MongooseRawResultQueryMiddleware`. I don't think its a good idea to inherit it because in case where we create a new type for middlewares that support some other option than that type can also contain middlewares from `MongooseRawResultQueryMiddleware` and vice versa. According to me, It's better if we maintain a separate type for this. Let me know your final decision and I'll update accordingly.

Reply:
i will be fine with it either way, @vkarpov15 what do you think?

Reply:
@hasezoey, once all the changes are completed and it's ready to be approved, there are conflicts of the branch merging process due to the latest pushes. I just wanted to know, do I have to resolve these conflicts, or will the reviewer handle it?

Reply:
I think avoiding inheriting is a reasonable idea.